File: benchmarks/py/map_hit_miss.py
import sys


def map_hit_miss(n: int) -> int:
  m = {i: i + 1 for i in range(n)}
  s = 0
  for i in range(n):
    k = i if i % 2 == 0 else i + n  # half misses
    s += m.get(k, 1)
  return s


if __name__ == "__main__":
  n = int(sys.argv[1]) if len(sys.argv) > 1 else 0
  print(map_hit_miss(n))

File: benchmarks/js/map_hit_miss.js
function mapHitMiss(n) {
  const m = {};
  for (let i = 0; i < n; i++) {
    m[i] = i + 1;
  }
  let sum = 0;
  for (let i = 0; i < n; i++) {
    const k = i % 2 === 0 ? i : i + n;  // half misses
    const v = m[k];
    sum += (v !== undefined ? v : 1);
  }
  return sum;
}

const n = process.argv[2] ? parseInt(process.argv[2]) : 0;
console.log(mapHitMiss(n));
